tests/dataframe/test_metrics_pytest.py

```diff
@@ -81,9 +81,10 @@ def test_flights_extended_metrics(self):
         logger.setLevel(logging.DEBUG)
 
         for func in self.extended_funcs:
-            pd_metric = getattr(pd_flights, func)(
-                **({"numeric_only": True} if func != "mad" else {})
-            )
+            if func == "mad":
```

Comment:
I noticed you introduced branching what if some day in the future we need to reimplement another function from scratch instead of creating a new branch every time can we extract this behavior out? perhaps have a utility class with our own custom implementation like

```
   customeFunctionMap = {"mad" : lambda x: (x - x.median()).abs().mean()}
```

and then dispatch it instead of branching for every new method we need to reimplent something like

if func in customFUnctionMap:
    // apply the custom function
else:
   // use the already given functionality.

Reply:
Done